TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
+                var lowPrice1 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[1];
+                var buyLimitPrice = lowPrice0 + (lowPrice1 - lowPrice0) / 2;
+                if(currentPrice != null)
+                {
+                    var tomorrowLowPrice = lowPrice0;
+                    if (tomorrowLowPrice > currentPrice.Low)
+                        buyLimitPrice = currentPrice.Low + (tomorrowLowPrice - currentPrice.Low) / 2;
+                    else
+                        buyLimitPrice = tomorrowLowPrice + (currentPrice.Low - tomorrowLowPrice) / 2;
+                }
+
+                growthRates.Add(new averageGrowthRate(symbol, growthRate, buyLimitPrice));
+            }
+        }
+
+        tradingActions.AddRange(sellActions);
+        tradingActions.AddRange(GetBuyActions(growthRates, cashAvailable, maxBuyCount, topGrowingSymbolsBuyRatio));
+
+        return tradingActions;
+    }
+
+    private async Task<DailyTradingData?> GetCurrentPrice(TradingSymbol symbol, CancellationToken token = default)
+    {
+        var today = DateOnly.FromDateTime(_clock.UtcNow.UtcDateTime);
+        var result = await _marketDataSource.GetDataForSingleSymbolAsync(symbol, today, today, token);
+
+        return (result == null) ? null : result.First();
+    }
+
+    private static bool IsPriceDecreasing(IReadOnlyList<decimal> closingPrices, int maxDaysDecreasing)
+    {
+        var isPriceDecreasing = true;
+        maxDaysDecreasing = maxDaysDecreasing > closingPrices.Count - 1 ? closingPrices.Count - 1 : maxDaysDecreasing;
+
+        for (int i = 1; i <= maxDaysDecreasing; i++)
+        {
+            if (closingPrices[i] > closingPrices[0])
+            {
+                isPriceDecreasing = false;
+                break;
+            }
+        }
+
+        return isPriceDecreasing;
     }
+
+    private static bool IsPriceIncreasing(IReadOnlyList<decimal> closingPrices, int maxDaysDecreasing)
+    {
+        var isPriceIncreasing = true;
+        maxDaysDecreasing = maxDaysDecreasing > closingPrices.Count - 1 ? closingPrices.Count - 1 : maxDaysDecreasing;
+
+        for (int i = 1; i <= maxDaysDecreasing; i++)
+        {
+            if (closingPrices[i] < closingPrices[i - 1])
+            {
+                isPriceIncreasing = false;
+                break;
+            }
+        }
+
+        return isPriceIncreasing;
+    }
+
+    private static decimal CalculateAverageGrowthRate(IReadOnlyList<decimal> closingPrices)
+    {
+        decimal totalGrowth = 0;
+
+        for (int i = 1; i < closingPrices.Count; i++)
+        {
+            decimal dailyGrowth = ((closingPrices[i] - closingPrices[i - 1]) / closingPrices[i - 1]) * 100;
+            totalGrowth += dailyGrowth;
+        }
+
+        decimal averageGrowth = totalGrowth / (closingPrices.Count - 1);
+
+        return averageGrowth;
+    }
```

Comment:
Maybe we could instead just use `(last - first) / first`? It's not exactly the same, but it's much simpler.